addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1954,7 +1954,10 @@ export class OdooEditor extends EventTarget {
             }
         }
         let insertedZws;
-        if (sel && !sel.isCollapsed && !range.startOffset && !range.startContainer.previousSibling) {
+        let { startContainer: start, startOffset, endContainer: end, endOffset } = range;
+        const [startBlock, endBlock] = [closestBlock(start), closestBlock(end)];
```

Comment:
As a rule, let's try and change the original commit as little as possible (to avoid unnecessary forward-port annoyances).

```suggestion
        const startBlock = closestBlock(start);
        const endBlock = closestBlock(end);
```